src/shared/components/common/markdown-textarea.tsx

```diff
@@ -575,11 +553,15 @@ export class MarkdownTextArea extends Component<
     this.setState({ languageId: val[0] });
   }
 
-  handleSubmit(i: MarkdownTextArea, event: any) {
+  async handleSubmit(i: MarkdownTextArea, event: any) {
     event.preventDefault();
     if (i.state.content) {
       i.setState({ loading: true, submitted: true });
-      i.props.onSubmit?.(i.state.content, i.state.languageId);
+      const success = await i.props.onSubmit?.(
+        i.state.content,
+        i.state.languageId,
+      );
+      i.setState({ loading: false, submitted: success ?? true });
```

Comment:
Shouldn't `onSubmit` always return a boolean, making the coalesce for `success` unnecessary?

Reply:
`onSubmit` is optional, which makes `success` of type `boolean | undefined`.